tests/third_party/cupy/manipulation_tests/test_kind.py

```diff
@@ -0,0 +1,134 @@
+import unittest
+
+import numpy
+import pytest
+
+import dpnp as cupy
+from tests.third_party.cupy import testing
+
+
+class TestKind(unittest.TestCase):
+    @pytest.mark.skip("dpnp.asarray_chkfinite() is not implemented yet")
+    @testing.for_orders("CFAK")
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_asarray_chkfinite(self, xp, dtype, order):
+        a = [0, 4, 0, 5]
+        return xp.asarray_chkfinite(a, dtype=dtype, order=order)
+
+    @pytest.mark.skip("dpnp.asarray_chkfinite() is not implemented yet")
+    @testing.for_orders("CFAK")
+    @testing.for_all_dtypes(no_bool=True)
+    def test_asarray_chkfinite_non_finite_vals(self, dtype, order):
+        a = [-numpy.inf, 0.0, numpy.inf, numpy.nan]
+        for xp in (numpy, cupy):
+            if xp.issubdtype(dtype, xp.integer):
+                error = OverflowError
+            else:
+                error = ValueError
+            with pytest.raises(error):
+                xp.asarray_chkfinite(a, dtype=dtype, order=order)
+
+    @testing.for_all_dtypes()
+    def test_asfarray(self, dtype):
+        a = cupy.asarray([1, 2, 3])
+        a_gpu = cupy.asfarray(a, dtype)
+        a_cpu = numpy.asfarray(a, dtype)
+        assert a_cpu.dtype == a_gpu.dtype

```

Comment:
need to handle devices without fp64 aspect differently:
```suggestion
        if (
            has_support_aspect64()
            or cupy.issubdtype(dtype, cupy.complexfloating)
            or cupy.issubdtype(dtype, cupy.floating)
        ):
            assert a_cpu.dtype == a_gpu.dtype
        else:
            assert a_cpu.dtype == cupy.float64
            assert a_gpu.dtype == cupy.float32
```